PayPal/src/main/java/com/braintreepayments/api/PayPalInternalClient.java

```diff
@@ -53,6 +47,17 @@ public void onResult(@Nullable final Configuration configuration, @Nullable Exce
                                         ? SETUP_BILLING_AGREEMENT_ENDPOINT : CREATE_SINGLE_PAYMENT_ENDPOINT;
                                 String url = String.format("/v1/%s", endpoint);
 
+                                String cancelUrl;
+                                String successUrl;
+                                if (payPalRequest.isAppLinkEnabled() && braintreeClient.getAppLinkReturnUri() != null) {
+                                    String appLink = braintreeClient.getAppLinkReturnUri().toString();
+                                    cancelUrl = String.format("%s/cancel", appLink);
+                                    successUrl = String.format("%s/success", appLink);
```

Comment:
Since we're appending to the merchants URL we'll have to make sure that they wildcard the URL they give us. Either that, or we ask them for explicit cancel and success full paths.

With the app switch project, we opted for asking them for a wildcarded path, specific to Braintree payments.

Reply:
Is there a wildcard format that we're currently using for the app switch integration? Is there a downside in just appending the cancel and success paths to the app link URL the merchant passes in?

The merchant is currently passing in a `Uri` type - I'm not sure if any URL validation is done when creating the Uri instance 🤔